tests/h/models/group_scope_test.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from sqlalchemy import inspect
+
+from h.models import GroupScope
+
+
+class TestGroupScope(object):
+    def test_save_and_retrieve_origin(self, db_session, factories):
+        origin = 'http://example.com'
+        factories.GroupScope(origin=origin)
+
+        group_scope = db_session.query(GroupScope).one()
+
+        assert group_scope.origin == origin
+
+    def test_subdomains_are_allowed_in_origin(self, db_session, factories):
+        factories.GroupScope(origin='http://www.example.com')
+        db_session.flush()
+
+    def test_port_is_allowed_in_origin(self, db_session, factories):
+        factories.GroupScope(origin='http://localhost:5000')
+        db_session.flush()
+
+    def test_there_is_no_validation_of_origin(self, db_session, factories):
+        factories.GroupScope(origin='diplodocus : 123')
+        db_session.flush()
```

Comment:
Maybe we'll want to add some validation, maybe even at the model level, at some point.